react-accordion-component/my-app/src/Accordion.js

```diff
@@ -0,0 +1,36 @@
+import './Accordion.css';
+
+/**
+ *
+ * @param {topics} An array of objects. Each object will contain a header and corresponding content for the Accordion panel.
+ */
+
+export default function Accordion ({topics, view, onSwapView}) {
+  return (
+    <div className="all-panels">
+      {topics.map((topic)=> (
+        <Topic className="panel"
+          view={view}
+          key={topic.header}
+          header={topic.header}
+          content={topic.content}
+          onSwapView={onSwapView} />))}
+    </div>
+  )
+}
+
+function Topic({view, header, content, onSwapView}){
+  return (
+    <>
+      <button className="panel-header"
+        type="button"
+        onClick={() => view===header ? onSwapView('none') : onSwapView(header)} >
```

Comment:
It would be best for Accordion to keep track of whether or not the contents should be drawn. The Topic shouldn't tell its parent whether to show or hide the contents, it should just tell the parent that it was clicked. Then let the parent decide what to do about the click.